src/main/java/duke/notes/Note.java

```diff
@@ -0,0 +1,209 @@
+package duke.notes;
+
+
+import duke.notes.budget.Budget;
+import duke.commands.CommandException;
+import duke.commands.NoteType;
+import duke.parser.DateException;
+import duke.ui.DukeUI;
+import java.util.Date;
+
+/**
+ * An abstract class that forms the basis of the various types of notes in {@code Duke}
+ * that extends the {@code Note} class. The class also provides common methods that would
+ * be used across these notes.
+ *
+ * @author tanqiuyu
+ * @since 2020-09-16
+ */
+public abstract class Note {
+
+    //VARIABLES-----------------------------------------
+    protected int serialNum;
+    protected String description;
+    protected Date addDate;
+    protected Date doneDate = null;
+    protected boolean isDone = false;
+
+
+    //CONSTRUCTORS--------------------------------------
+    /**
+     * This method is used to initialise the attributes common to the various type of notes
+     * that extends the {@code Note} class.
+     *
+     * @param serialNum The serial number automatically assigned for identification purposes.
+     * @param description The description of the note.
+     * @param addDate The date and time the note was added.
+     */
+    public Note(int serialNum, String description, Date addDate) {
+
+        assert !description.isEmpty() && !description.isBlank() : "Note description cannot be empty or blank.";
+        this.serialNum = serialNum;
+        this.description = description;
+        this.addDate = addDate;
+    }
+
+    /**
+     * This method is used to initialise the various type of notes
+     * that extends the {@code Note} class.
+     */
+    @SuppressWarnings("unused")
+    public Note() {}
+
+
+    //SET STATEMENTS------------------------------------
+    /**
+     * This method is used to mark an outstanding {@code Note} object as completed.
+     *
+     * @param doneDate The date and time the {@code Note} had concluded.
+     * @return boolean True if the operation is successful.
+     * @exception CommandException If there are errors in the command input.
+     * @exception DateException If there are errors in the formats or substance of {@code Date} objects.
+     */
+    public boolean markAsDone(Date doneDate) throws CommandException, DateException, InterruptedException {
+        DukeUI.printDivider();
+        if(this.isDone) {
+            System.out.println("    Note #" + this.serialNum + " was already done!");
+            return false;
+        } else {
+            this.isDone = true;
+            this.doneDate = doneDate;
+            System.out.println("    Noted! I've marked Note #" + this.serialNum + " as done.");
+            return true;
+        }
+    }
```

Comment:
No need for else here as well
```java
public boolean markAsDone(Date doneDate) throws CommandException, DateException, InterruptedException {
        DukeUI.printDivider();
        if(this.isDone) {
            System.out.println("    Note #" + this.serialNum + " was already done!");
            return false;
        }
        this.isDone = true;
        // ...
    }
```

Reply:
Noted Prof, I've also looked thru the rest of the code for similar instances.